server/client/ruby/candlepin_api.rb

```diff
@@ -1227,6 +1236,11 @@ def post_text(uri, data=nil)
     return response.body
   end
 
+  def post_text_return_json(uri, data=nil)
```

Comment:
What's the benefit or difference for this method that requires a new mechanism for posting json?